plugins/pencilblue/controllers/actions/salesforce/login_callback.js

```diff
@@ -38,10 +38,33 @@ module.exports = function LoginSalesforceCallbackControllerModule(pb) {
                     pb.log.warn('Something went wrong during the state parsing: ', e);
                 }
             }
-            let redirectLocation = await this.getRedirectLink(response, state);
+            if (response.updateEmail) {
+                await this.updateEmail(response.user);
+            }
+            let redirectLocation = await this.getRedirectLink(response.user, state);
             this.redirect(redirectLocation, cb);
         }
 
+        async updateEmail(user) {
+            try {
+                this.identityProviderService = this.createService('IdentityProviderService', 'tn_auth');
+                this.encryptProfileIdService = this.createService('EncryptProfileIdService', 'tn_profile');
+                // update user, jobseeker_profile collections and matrix jobseeker_profile table
+                const siteQueryService = new pb.SiteQueryService();
+                const jobseekerProfile = await siteQueryService.loadByValuesAsync({
+                    externalUserId: user.external_user_id,
+                    object_type: 'jobseeker_profile'
+                }, 'jobseeker_profile');
+                let encryptedProfileDID = null;
+                if (jobseekerProfile) {
+                    encryptedProfileDID = await this.encryptProfileIdService.encrypt(jobseekerProfile.profileDID);
```

Comment:
...so you would need to check this.encryptProfileIdService exists before calling it